### PR TITLE
Deserialize Sources into the right SourceKind

### LINKED ISSUE #5: PublishedRepository instances have incorrect source types
PublishedRepository has a SourceKind and a Sources array, latter is always deserialized into a Repository. It should however be a Snapshot when kind is snapshot

```
=> #<Aptly::PublishedRepository Architectures=["all", "amd64", "i386", "source"], Distribution="xenial", Label="KDE neon - Dev Edition Stable", Origin="neon", Prefix="dev/stable", SkipContents=false, SourceKind="snapshot", Sources=[#<Aptly::Repository Component="main", Name="stable_xenial_before_qt56">], Storage="">
```

probably would be good to run SourceKind through a case-when and raise if SourceKind is unsupported by the gem

### PR DESCRIPTION
**Pull request by hsitter**
Fixes #5

### CODE REVIEW

File: lib/aptly/publish.rb [hunk 1]
```diff
@@ -1,14 +1,29 @@
 require_relative 'representation'
 
 module Aptly
+  module Errors
```
Comment (hsitter) [inline lib/aptly/publish.rb]: This should be in errors.rb

File: lib/aptly/publish.rb [hunk 2]
```diff
@@ -1,14 +1,29 @@
...
+  end
+end
+
+module Aptly
   # A published repository representation.
   # Published repositories are not {Repository} instances as they are in fact
   # comprised of one or more different repositories.
   # @see http://www.aptly.info/doc/api/publish/
   class PublishedRepository < Representation
     def initialize(*args)
```
Comment (hsitter) [inline lib/aptly/publish.rb:9]: This method is now too long. needs splitting

File: test/publish_test.rb [hunk 3]
```diff
@@ -39,4 +39,36 @@ def test_update!
...
+      'Distribution' => 'distro',
+      'Label' => '',
+      'Origin' => '',
+      'Prefix' => 'prefix/kewl-repo-name',
+      'SourceKind' => 'snapshot',
+      'Sources' => [{ 'Component' => 'main', 'Name' => 'kitten' }],
+      'Storage' => ''
+    )
+
+    assert(pub.Sources[0].is_a?(Aptly::Snapshot))
```
Comment (hsitter) [inline test/publish_test.rb:56]: There should also be a Repository source and an assertion on it to make sure all 3 types actually work. As well as the combination of the two valid types.

**Comment by hsitter** [comment]
Integration test for this is missing.

File: test/publish_test.rb (hunk 3 above)
Comment (shadeslayer) [inline test/publish_test.rb:56]: Does each source list it's kind when SourceKind is a combination of two valid types?

Comment (hsitter) [inline test/publish_test.rb:56]: Nope, so you need a third test.

Comment (shadeslayer) [inline test/publish_test.rb:56]: Can you point me to a example of what this looks like? I can't find anything in the documentation.

Comment (shadeslayer) [inline test/publish_test.rb:56]: AFAICT from https://www.aptly.info/doc/api/publish/ , SourceKind is always only a string.

Comment (hsitter) [inline test/publish_test.rb:56]: Ah yes, nevermind then.

**Comment by shadeslayer** [comment]
Should I go ahead and rebase this onto master then?

**Comment by shadeslayer** [comment]
Merged

**Pull request closed without merging**